### ISSUE TITLE
Hanami commands (server/console) don't print error backtrace if the error occured during Hanami boot and code reloading is disabled

### ISSUE THREAD

**Issue by smaximov** [hanami/hanami#890]
**UPDATE 1**: to reproduce this issue, you need to disable code reloading.

This makes it very difficult to debug such errors.

Consider this snippet

``` ruby
module MySchema
  class UserRepository < Hanami::Repository
  end
end
```

If I start any Hanami command, it will fail with the following error:

```
invalid attribute name `my_schema/users'
```

Backtrace is not printed, so the user need to guess where to start to debug this error.

This is just more recent (and actually straightforward to debug) example that I've stumbled upon, it can also be a message like "no implicit conversion of nil into String (TypeError)" or something similar which gives little to no information about the reason behind the error.

The cause is the following code:

https://github.com/hanami/hanami/blob/8b0fb2e5d89c8f9ff6d3cc71851c99e29e4f9dd4/lib/hanami/cli/commands/command.rb#L86-L89

So the workaround is to delete the `rescue` clause or re-raise the exception or print the backtrace manually.

**Comment by AlfonsoUceda** [comment]
@smaximov thanks for the issue, I agree with this and we should print the backtrace.

**Comment by AlfonsoUceda** [comment]
@smaximov could you provide a description to know how to reproduce the error? trying with a fresh application and adding that snippet it didn't work.

**Comment by smaximov** [comment]
@AlfonsoUceda, seems like it depends on whether code reloading is enabled. This can be done with a [fresh hanami app](https://github.com/smaximov/commands_load_issue) (the only code I've added is [this](https://github.com/smaximov/commands_load_issue/blob/0ca2d78a44c5d2d2c265652fbbd2846f8c23502d/lib/commands_load_issue/repositories/my_schema/user_repository.rb) snippet). You need to remove `shotgun` from the project or start the server with `--no-code-reloading`.

I didn't notice it because my project doesn't use code reloading, so I apologize for the confusion. I've updated the issue.

**Comment by AlfonsoUceda** [comment]
@smaximov thanks! I'll test as soon as possible ;)

**Comment by cllns** [comment]
#900 might fix this?

**Comment by AlfonsoUceda** [comment]
@cllns yep

**Comment by jodosha** [comment]
Implemented by #900
